[https://nvbugs/5394409][feat] Support Mistral Small 3.1 multimodal in Triton Backend

dbari opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added end-to-end support for the “pixtral” multimodal model.
  - Accept multiple images via new optional image inputs; supports URLs, base64, and local files.
  - Dynamic image size handling in preprocessing and encoders.

- Configuration
  - Prompt embedding table data type is now configurable.
  - Added model path parameter for Hugging Face assets.

- Tools
  - Multimodal client supports “pixtral” and raw image-bytes input.

- Tests
  - Added integration tests and engine preparation for Mistral Small 3.1 Pixtral.

- Chores
  - Enabled Git LFS for test image assets.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

## Description

Implemented Mistral Small 3.1 in Triton backend. It supports multiple images for each prompt and the images can have different resolution. Each image will be embedded into the prompt at the location of its respective `[IMG]` placeholder and have a length of one token for every 28x28 patch (disregarding padding and line break / end of image tokens). Also, since the vision encoder has a fixed `max_batch_size` but the request might contain an arbitrary number of images, the images are processed in batches of size `max_batch_size`.

After discussing with @Tabrizian (thanks!), I understood the interaction between the number of requests (triton batch) and the batch size of each request. The code has been reworked to allow an arbitrary triton batch size (number of requests per `execute()` call) ~and batch size per request~ but the batch size in each request should be 1.

Side note: The model runs successfully only when using the `python` backend for the `tensorrt_llm` step in the ensemble. This seems to be a bug or difference in configuration in the C++ `tensorrtllm` backend, that is independent of this PR. Unless there is another suggestion, I will open an issue for it.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

dbari committed:
Make type of prompt_embedding_table configurable

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Implement Mistral Small 3.1 in triton_backend

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Add support for Mistral Small 3.1 (pixtral) to multimodal client

- Serialize multiple images that are decoded by the server

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Pre-commit fixes

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Fix max_num_images

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Ignore AttributeError for other models

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Made hf_model_path configurable

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Fix CodeRabbit review comments

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #14492](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14492/) [ run ] triggered by Bot

dbari committed:
Add substitution for prompt_embedding_table_data_type to tests

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Add test for Mistral Small 3.1

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Test Mistral Small 3.1 with multiple input images

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Fix behavior when [IMG] placeholder(s) missing

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Add tests to test_triton.py

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #14624](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14624/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #11048 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11048/) completed with status: 'FAILURE'

Tabrizian commented on triton_backend/all_models/inflight_batcher_llm/preprocessing/config.pbtxt:
```diff
@@ -58,6 +58,13 @@ input [
     },
+    # An arbitrary number of images for pixtral
+    {
+        name: "IMAGES"
```
why can't we use the "IMAGE_BYTES" input?

dbari replied:
We could use it, however in this case we would need to pad any smaller images and we'd need to define a new `IMAGE_SIZES` input for the ensemble, so that the preprocessing knows how big each of the images is and how many tokens to allocate for it. If you prefer it this way, I can gladly do it, I have no strong feelings either way. The reason I opted for this implementation is that I consider it more efficient (no padding of images over the network and less data overall, as the decoding takes place on the server)

dbari committed:
PR finding: Use tokenizer_dir instead of defining new hf_model_path

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented on triton_backend/all_models/inflight_batcher_llm/preprocessing/1/model.py:
```diff
@@ -151,10 +151,23 @@ def initialize(self, args):
+                    'mllama', 'llava_onevision', 'qwen2_vl', 'pixtral'
+            ]:
+                # Read the correct path from the config
+                hf_model_path = model_config['parameters']['hf_model_path'][
```
Done in f109060

jaedeok-nvidia reviewed (commented):
Thanks a lot for adding Pixtral support to the trtllm backend of Triton. I’ve left several comments. Could you please take a look?

dbari commented on triton_backend/all_models/multimodal/multimodal_encoders/config.pbtxt:
```diff
@@ -76,7 +76,7 @@ input [
         name: "image_sizes"
         data_type: TYPE_INT64
-        dims: [ 2 ]
+        dims: [ -1, 2 ]
```
Please see [my comment](https://github.com/NVIDIA/TensorRT-LLM/pull/6714#discussion_r2265896849) above. Since this came up twice already, I will go on and change it as @Tabrizian and you proposed.

dbari commented on triton_backend/all_models/multimodal/ensemble/config.pbtxt:
```diff
@@ -57,6 +57,13 @@ input [
+      data_type: TYPE_STRING
+      dims: [ -1 ]
+      optional: true
+  },
```
Please see [my comment](https://github.com/NVIDIA/TensorRT-LLM/pull/6714#discussion_r2265896849) above. Since this came up twice already, I will go on and change it as @Tabrizian and you proposed.

jaedeok-nvidia replied:
Oh, I missed. Increasing model-specific field will confuse users.  Thanks for addressing it. Please let us know once it updated.

dbari replied:
Removed the new input for images. However, I added a new input `image_sizes_input` instead, because the images are passed through `image_bytes_input` padded to the size of the largest and the preprocessing needs to know their original sizes before padding.

Marking this as resolved. If more discussion is needed please reopen or open a new thread.

dbari commented on triton_backend/all_models/inflight_batcher_llm/preprocessing/1/model.py:
```diff
@@ -716,10 +746,8 @@ class VisionPreProcessor:
-                 vision_model_processor,
-                 preprocessor_model_config={}):
+    def __init__(self, vision_model_type, vision_model_processor,
+                 preprocessor_model_config, hf_config):
```
This constructor is only used in `TritonPythonModel.initialize()` in the same file [here](https://github.com/dbari/TensorRT-LLM/blob/feature/5394409-mistral-small-3.1/triton_backend/all_models/inflight_batcher_llm/preprocessing/1/model.py#L163) and always supplies the `preprocessor_model_config` argument, so I removed the default value (also seeing it was a mutable `{}`)

I did not consider it to be part of a public API, as it cannot be called from anywhere else, however if you want to keep the API stable, I can add the default value back as `None`

jaedeok-nvidia replied:
Thanks for the clarification. These files are provided as examples so that users can customize them, rather than treating them as part of the  public API. I was concerned about that. However, if this constructor is only called internally, I think it’s fine to update it if it helps make the example clearer. So, let's keep the current version of this PR.

dbari commented on triton_backend/all_models/multimodal/multimodal_encoders/1/model.py:
```diff
@@ -276,6 +291,29 @@ def get_requests(self, request: List) -> Dict[str, torch.Tensor]:
+            assert batch_size == 1, "Only support batch size 1 for Pixtral, because each batch can contain a different number of images"
+            d_min = torch.finfo(self.vision_output_dtype).min
+            total_images = img_tensor.shape[0] * img_tensor.shape[1]
+            num_patches = self.image_size // self.patch_size
```
Yes, it's square in this model, as padded by the preprocessing [here](https://github.com/dbari/TensorRT-LLM/blob/bf1deadb7b2bc39efd7e1fb3c18d00403f7fd128/triton_backend/all_models/inflight_batcher_llm/preprocessing/1/model.py#L1096). The full shape is `[batch_size, images, 3, 1540, 1540]`

jaedeok-nvidia replied:
Thanks, then can I know what the `total_images` indicates? Total number of images across given batch? If then, for a given batch, should the number of images be identical to `images`?

jaedeok-nvidia replied:
There is `batch_size == 1` in the above, I missed. Thanks for clarification.

dbari committed:
Fix dimensions of image_sizes for llava_onevision

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
PR findings, fixes, enhancements

- Remove extra input images_input, add image_sizes_input (only for pixtral)
- Fix case of using multimodal model with text-only prompt
- Implement inference for higher Triton batch size and request batch size
- Implement vision encoder inference for arbitrary number of images
- Add [INST] and [/INST] placeholders automatically if not present

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15010](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15010/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #11337 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11337/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #15091](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15091/) [ run ] triggered by Bot

dbari committed:
Fix PR findings by CodeRabbit

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented on tests/integration/defs/triton_server/build_engines.py:
```diff
@@ -1763,3 +1763,72 @@ def prepare_rcca_nvbug_4714193_engine(tensorrt_llm_example_root,
+        "--max_batch_size=8",
+        "--max_input_len=4096",
+        "--max_seq_len=4096",
+        "--max_multimodal_len=32768",
+        "--use_paged_context_fmha=enable",
```
Fixed in c54423839fe59114a528b986a100e69aec8bcc2f

dbari commented on tests/integration/defs/triton_server/build_engines.py:
```diff
@@ -1763,3 +1763,72 @@ def prepare_rcca_nvbug_4714193_engine(tensorrt_llm_example_root,
+        print_info(f"Skipped: {convert_cmd}")
+        print_info(f"Skipped: {build_cmd}")
+        print_info(f"Skipped: {build_visual_engine_cmd}")
+
```
Fixed in c54423839fe59114a528b986a100e69aec8bcc2f

dbari commented on tests/integration/defs/triton_server/build_model.sh:
```diff
@@ -699,4 +699,33 @@ if [ "$MODEL" = "qwen2_vl" ]; then
+    echo "Install pixtral requirements if any"
+    # Reuse general requirements (multimodal run path supports pixtral)
+    pip install -r $LLM_BACKEND_ROOT/requirements.txt || true
+
```
Fixed in c54423839fe59114a528b986a100e69aec8bcc2f

dbari commented on triton_backend/all_models/inflight_batcher_llm/preprocessing/1/model.py:
```diff
@@ -136,9 +136,9 @@ def initialize(self, args):
             assert self.model_type in [
-                'llava', 'blip2-opt', 'vila', 'mllama', 'llava_onevision',
-                'qwen2_vl'
-            ], f"[TensorRT-LLM][ERROR] Currently supported multi-modal models are llava, blip2-opt, vila, mllama, llava_onevision and qwen2_vl. Got {self.model_type}."
+                'llava', 'blip2-opt', 'pixtral', 'vila', 'mllama',
+                'llava_onevision', 'qwen2_vl'
+            ], f"[TensorRT-LLM][ERROR] Currently supported multi-modal models are llava, blip2-opt, pixtral, vila, mllama, llava_onevision and qwen2_vl. Got {self.model_type}."
 
```
Won't fix due to consistency, as assertions were already used throughout the code.

dbari committed:
Add type hints to new functions

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15091](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15091/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11394 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11394/) completed with status: 'FAILURE'

dbari committed:
Implement changes for Mistral Small 3.1 in tensorrt_llm_bls

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Check output of tests

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Add two of the tests to l0_a100.yml

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented on tests/integration/defs/triton_server/test_triton_llm.py:
```diff
@@ -4020,19 +4012,54 @@ def test_mistral_small_3_1_24b_pixtral(
+        {
+            "text": "In as few words as possible, what sports are depicted in the images?",
+            "image": "https://images.pexels.com/photos/32285228/pexels-photo-32285228.jpeg?cs=srgb&fm=jpg&w=1280&h=720," \
+                     "https://images.pexels.com/photos/8975010/pexels-photo-8975010.jpeg?cs=srgb&fm=jpg&w=640&h=960",
```
Is it ok to use external images like this? Personally, I would prefer to have a few of them in the repository or in an internal location instead of downloading them from an external server on each run, however the same is done elsewhere with `merlion.png`. Any suggestions?

dbari committed:
Fix the case of no image present

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
Fix engine max_input_length and enable tests for BLS

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15265](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15265/) [ run ] triggered by Bot

dbari commented:
/bot run --stage-list "A30-Triton-Post-Merge-1, A30-Triton-Post-Merge-2, B200_PCIe-Triton-Post-Merge-1, A100X-Triton-Post-Merge-1, A100X-Triton-Post-Merge-2"

dbari committed:
Bugfix for triton_server/test_triton.py::test_gpt_gather_logits

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15279](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15279/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #15279](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15279/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #11534 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11534/) completed with status: 'FAILURE'

dbari committed:
Implement MockTritonTensor.to_dlpack()

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented:
/bot run --stage-list "A30-Triton-Post-Merge-1, A30-Triton-Post-Merge-2"

dbari committed:
Bugfix for triton_server/test_triton.py::test_python_preproc_unit_tests

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15297](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15297/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11548 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11548/) completed with status: 'FAILURE'

dbari commented:
/bot run --stage-list "A30-Triton-Post-Merge-1, A30-Triton-Post-Merge-2"

tensorrt-cicd commented:
[PR_Github #15320](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15320/) [ run ] triggered by Bot

Tabrizian commented on triton_backend/all_models/multimodal/multimodal_encoders/1/model.py:
```diff
@@ -530,17 +662,67 @@ def postprocess_output_tensors(
+        embeddings = []
+        for start_idx in range(0, batch_size, self.vision_max_batch_size):
+            end_idx = min(start_idx + self.vision_max_batch_size, batch_size)
+            logger.info(
```
```suggestion
            logger.verbose(
```

To avoid print statements for every request.

Tabrizian commented:
@dbari thanks for the PR.

> The model runs successfully only when using the python backend for the tensorrt_llm step in the ensemble. This seems to be a bug or difference in configuration in the C++ tensorrtllm backend, that is independent of this PR. Unless there is another suggestion, I will open an issue for it.

What is the issue? Are we able to get correct outputs outside Triton?

dbari committed:
PR finding: Mutable default parameter

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented on triton_backend/all_models/inflight_batcher_llm/preprocessing/1/model.py:
```diff
@@ -747,8 +747,11 @@ class VisionPreProcessor:
+    def __init__(self,
+                 vision_model_type,
+                 vision_model_processor,
+                 preprocessor_model_config={},
```
Oops, I reverted to the original state of the code and it slipped by me, good catch.
Fixed in a76e1081cbc8e1ceff60bc1517f0e8ea03d93913

dbari commented on triton_backend/all_models/inflight_batcher_llm/preprocessing/config.pbtxt:
```diff
@@ -58,6 +58,13 @@ input [
     },
+    # Required for pixtral
+    {
+        name: "IMAGE_SIZES"
```
It would be possible for `IMAGE_URL` since the backend will decode the images. For `IMAGE_BYTES` it's not possible, unless we encode the size into the content in some way, e.g. automatically crop black borders from the bottom/right for all images. But this might interfere with the user actually providing such an image and wanting to infer with it. While it could be done, I find it is too fragile and introduces a surprise factor,

By the way I saw that `IMAGE_URL` has `dims: [ 1 ]`, I'll change to `dims: [ -1 ]` to support multiple images. The `client.py` does not support URLs, so this path isn't really tested.

dbari committed:
PR finding: remove unused self.image_token_index

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented on triton_backend/all_models/multimodal/multimodal_encoders/1/model.py:
```diff
@@ -162,6 +164,21 @@ def initialize(self, args):
+                self.image_size = hf_config.vision_config.image_size
+                self.patch_size = hf_config.vision_config.patch_size
+                self.vocab_size = hf_config.text_config.vocab_size
+                self.image_token_index = hf_config.image_token_index
```
Removed in e61a0ea29d2aab73982d98614414be7eebe36448

dbari committed:
Make logging for vision encoder batches less verbose

Co-authored-by: Iman Tabrizian <10105175+Tabrizian@users.noreply.github.com>
Signed-off-by: Dimitrios Bariamis <dbari@users.noreply.github.com>

dbari committed:
Change logger.verbose() to logger.debug()

- The logger has no method logger.verbose()

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari committed:
PR findings: Remove test from test_triton.py and add more tests to list

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

jaedeok-nvidia reviewed (approved):
Thanks for adding the Mistral model to the triton backend. Looks good to me if the CI passes.

dbari committed:
Allow multiple images per batch via IMAGE_URL

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented:
/bot run

dbari commented:
Getting unrelated error `ImportError: libnccl.so.2: cannot open shared object file: No such file or directory` in CI. Rebasing to current main and rebuilding.

tensorrt-cicd commented:
[PR_Github #15744](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15744/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #15744](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15744/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11833](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11833/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #15790](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15790/) [ run ] triggered by Bot

dbari committed:
Use default backend (tensorrtllm) for test_mistral_small_3_1_24b_pixtral

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented:
/bot run

tensorrt-cicd commented:
[PR_Github #15875](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15875/) [ run ] triggered by Bot

dbari committed:
Change test_mistral_small_3_1_24b_pixtral to use local files (LFS) instead of URLs

Signed-off-by: Dimitrios Bariamis <12195802+dbari@users.noreply.github.com>

dbari commented:
/bot run --stage-list "A100X-Triton-Post-Merge-1, A100X-Triton-Post-Merge-2"

tensorrt-cicd commented:
[PR_Github #16054](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/16054/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #16054](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/16054/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #12073 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/12073/) completed with status: 'SUCCESS'
